### DISCUSSION TITLE
4.0.0 - Coming Soon

### DISCUSSION THREAD

**Discussion by jsjoeio** [coder/code-server#4490] [General]
Hi all!

As you may have noticed, we haven't had a release since Sep. 15.

We've been going through a major rewrite of code-server's server + rearchitecting how code-server uses VS Code under the hood. We made a huge step forward (big thanks to @TeffenEllis & @code-asher) with [a PR](https://github.com/cdr/code-server/pull/4414) we merged yesterday.

While this is brings us closer to a release, we identified a number of features/bugs we need to patch now that we're mostly using the upstream server. We've listed them all out in separate issues and are tracking in [this milestone](https://github.com/cdr/code-server/milestone/32).

Once we finish all those, we'll be able to release 4.0.0. 

Thanks for your continued support of the project, your help and your patience as we get next version of code-server out! 🙏

Feel free to drop any questions, comments or concerns below.

**Comment by lg-nineoneninethreenine** [comment]
So, till then, 3.12.1, and 3.12.2, right?

**Reply by jsjoeio** [reply]
Nope! Because this next release will have breaking changes, we are going straight to 4.0.0 with the next release so there won't be a 3.12.1 or 3.12.2.

**Comment by seohayeon** [comment]
Is nodejs v16 supported?

**Reply by lg-nineoneninethreenine** [reply]
Hmm... Not yet but we are ready to add Node 16 support here, if microsoft/vscode adds support, then we can also add support soon.

**Reply by jsjoeio** [reply]
Yeah, echoing @im-coder-lg it doesn't seem like it's possible at the moment. See this issue: https://github.com/cdr/code-server/issues/4286
